src/saltext/vmware/modules/esxi.py

```diff
@@ -1092,6 +1093,521 @@ def remove_user(
+            h.configManager.virtualNicManager.DeselectVnicForNicType(service, vmk_device)
+
+    vsan_config = vim.vsan.host.ConfigInfo()
+    vsan_config.networkInfo = h.configManager.vsanSystem.config.networkInfo
+    current_vsan_vnics = [
+        portConfig.device for portConfig in h.configManager.vsanSystem.config.networkInfo.port
+    ]
+    if enable_vsan:
+        if vmk_device not in current_vsan_vnics:
+            vsan_port_config = vim.vsan.host.ConfigInfo.NetworkInfo.PortConfig()
+            vsan_port_config.device = vmk_device
+            if vsan_config.networkInfo is None:
+                vsan_config.networkInfo = vim.vsan.host.ConfigInfo.NetworkInfo()
+                vsan_config.networkInfo.port = [vsan_port_config]
+            else:
+                vsan_config.networkInfo.port.append(vsan_port_config)
+    elif enable_vsan is False and vmk_device in current_vsan_vnics:
+        vsan_config.networkInfo.port = list(
+            filter(lambda portConfig: portConfig.device != vmk_device, vsan_config.networkInfo.port)
+        )
+    task = h.configManager.vsanSystem.UpdateVsan_Task(vsan_config)
+    utils_common.wait_for_task(task, h.name, "UpdateVsan_Task")
+    return vmk_device
+
+
+def get_vmkernel_adapters(
+    adapter_name=None,
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    Update VMKernel Adapter on matching ESXi hosts.
+
+    adapter_name
+        Filter by this vmkernel adapter name.
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.get_vmkernel_adapter port_group_name=portgroup1
+    """
+    log.debug("Running vmware_esxi.get_vmkernel_adapter")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        for h in hosts:
+            vmk_devices = []
+            for v in h.config.network.vnic:
+                if adapter_name and v.device != adapter_name:
+                    continue
+                vmk_devices.append(v.device)
+            ret[h.name] = vmk_devices
+        return ret
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+        vim.fault.AlreadyExists,
+    ) as exc:
```

Comment:
**suggestion(non-blocking)** With the several errors, repeated in most/all of these, this could be a toplevel collection:

```
DEFAULT_EXCEPTIONS = (vim.fault.InvalidState, ...)

...

try:
    ...
except DEFAULT_EXCEPTIONS as exc:
    raise ...
```    

would simplify some of these a bit.

Reply:
(I'm not a terrible fan of "DEFAULT_EXCEPTIONS" but my brain is done with clever names for the day apparently)

Reply:
will change